src/command.coffee

```diff
@@ -102,7 +102,19 @@ exports.run = ->
   process.argv = process.argv[0..1].concat literals
   process.argv[0] = 'coffee'
 
-  opts.output = path.resolve opts.output  if opts.output
+  if opts.output
+    outputBasename = path.basename opts.output
+    if '.' in outputBasename and
+       not helpers.ends(opts.output, path.sep) and
+       not /^\.+$/.test(outputBasename) # Not `.` or `..`.
```

Comment:
Three or more dots is a valid file name, though. We should only special case `.` and `..`.